jta/src/main/java/io/smallrye/context/jta/context/propagation/JtaContextProvider.java

```diff
@@ -117,17 +118,23 @@ private TransactionManager tm() {
         if (lifecycleManagers.isResolvable()) {
             lifecycleManagers.get().setProvider(this);
         }
-        tm = CDI.current().select(TransactionManager.class).get();
-        if (tm != null) {
-            // validate it, because it can be an empty proxy
-            try {
-                tm.getStatus();
-            } catch (CreationException | SystemException x) {
-                x.printStackTrace();
-                // not really there
-                transactionManagerNotAvailable = true;
-                return null;
+        try {
+            tm = CDI.current().select(TransactionManager.class).get();
+            if (tm != null) {
+                // validate it, because it can be an empty proxy
+                try {
+                    tm.getStatus();
+                } catch (CreationException | SystemException x) {
+                    x.printStackTrace();
+                    // not really there
+                    transactionManagerNotAvailable = true;
+                    return null;
+                }
             }
+        } catch (UnsatisfiedResolutionException x) {
```

Comment:
The unsatisfied resolution would come from the very first line - `CDI.current().select(TransactionManager.class).get();`
Namely from the `get()` part. You can check for it before it throws and avoid the try-catch altogether. The `Instance` class has methods such as `isResolvable()`.
Same logic is actually used few lines above for `LifecycleManager` resolution.

Reply:
Yeah, I did wonder about that. Do you think it's better?

Reply:
Yea, it is certainly cleaner than try-catch blocks ;-)

Reply:
OK done!